spring-context/src/main/java/org/springframework/validation/DataBinder.java

```diff
@@ -546,9 +550,11 @@ private void assertValidators(Validator... validators) {
 	 * @see #setValidator(Validator)
 	 * @see #replaceValidators(Validator...)
 	 */
-	public void addValidators(Validator... validators) {
+	public void addValidators(@Nullable Validator... validators) {
```

Comment:
Here it seems sufficient to mark `setValidator` as nullable, paired with `getValidator()`. The multi-validator case in `getValidators()` isn't nullable anyway, so `addValidators` doesn't have to be nullable either (aside from not being setter and differing in array vs list as well).